nodejs/eks/cluster.ts

```diff
@@ -627,6 +624,74 @@ export function createCore(
         },
     );
 
+    pulumi.output(args.fargate).apply((fargate) => {
+        if (args.kubeProxyAddonOptions?.enabled ?? true) {
+            const kubeProxyVersion: pulumi.Output<string> = args.kubeProxyAddonOptions?.version
+                ? pulumi.output(args.kubeProxyAddonOptions?.version)
+                : aws.eks
+                      .getAddonVersionOutput(
+                          {
+                              addonName: "kube-proxy",
+                              kubernetesVersion: eksCluster.version,
+                              mostRecent: true, // whether to return the default version or the most recent version for the specified kubernetes version
+                          },
+                          { parent, provider },
+                      )
+                      .apply((addonVersion) => addonVersion.version);
+
+            const kubeProxyAddon = new aws.eks.Addon(
+                `${name}-kube-proxy`,
+                {
+                    clusterName: eksCluster.name,
+                    addonName: "kube-proxy",
+                    resolveConflictsOnCreate:
+                        args.kubeProxyAddonOptions?.resolveConflictsOnCreate ?? "OVERWRITE",
+                    resolveConflictsOnUpdate:
+                        args.kubeProxyAddonOptions?.resolveConflictsOnUpdate ?? "OVERWRITE",
+                    addonVersion: kubeProxyVersion,
```

Comment:
we should set the `preserve` property to `true`. Otherwise this would delete the resources from the cluster when users toggle the `enabled` setting

Reply:
That brings up a question - Is enabled a thing that only works on initial deployment?

Reply:
The addons can be created/deleted at any time

Reply:
Updated